### PR TITLE
Add job list command

### PR DESCRIPTION
**Pull request by IrvingMg**
## Fixes / Features
- Add job list command

## Testing / Documentation
Testing details.

- [ y/n ] Tests pass
- [ y/n ] Appropriate changes to documentation are included in the PR

### CODE REVIEW

File: src/xpk/parser/job.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  )
+
+  ### "job ls" command parser ###
+  job_list_parser = job_subcommands.add_parser('ls', help='List Slurm jobs.')
+
+  job_list_optional_arguments = job_list_parser.add_argument_group(
+      'Optional Arguments', 'Arguments optional for job list.'
+  )
+
+  job_list_optional_arguments.add_argument(
```
Comment (pawloch00) [inline src/xpk/parser/job.py:38]: I don't think there is possiblity to definie custom namespace in xpk, but it needs to be checked.

Comment (IrvingMg) [reply src/xpk/parser/job.py:38]: That's correct. Setting up the default namespace to something other than “default” will be provided in a separate task.

Comment (pawloch00) [reply src/xpk/parser/job.py:38]: But this namespace should be created somewhere  and I don't think it is possible now. (I don't know about such plans too :P)

Comment (IrvingMg) [reply src/xpk/parser/job.py:38]: @mwysokin would you mind confirming that namespace configuration is a future feature that will be required?

Comment (pawloch00) [reply src/xpk/parser/job.py:38]: After consultation I --namespace option should be removed now

Comment (IrvingMg) [reply src/xpk/parser/job.py]: Done.

**Comment by IrvingMg** [comment]
> Can you please add tests to .github workflows including your change?

Done.

File: .github/workflows/build_tests.yaml [hunk 2]
```diff
@@ -88,6 +88,10 @@ jobs:
       run: python3 xpk.py workload delete --workload $WORKLOAD_NAME --cluster $TPU_CLUSTER_NAME --zone=us-central2-b
     - name: Delete the Pathways workload on the cluster
       run: python3 xpk.py workload delete --workload $PATHWAYS_WORKLOAD_NAME --cluster $TPU_CLUSTER_NAME --zone=us-central2-b
+    - name: Run a batch job on the cluster
+      run: python3 xpk.py batch --cluster $TPU_CLUSTER_NAME --zone=us-central2-b test.sh
+    - name: List out the Slurm jobs on the cluster
+      run: python3 xpk.py job ls
```
Comment (pawloch00) [inline .github/workflows/build_tests.yaml:94]: please add an assert(grep) to check output sanity

File: src/xpk/commands/job.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+
+def job_list(args) -> None:
+  """Function around job list.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    0 if successful and 1 otherwise.
```
Comment (mbobrovskyi) [inline src/xpk/commands/job.py:32]:
```suggestion
    None
```

File: src/xpk/commands/job.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  """
+  add_zone_and_project(args)
+  xpk_print(f'For project {args.project} and zone {args.zone}:', flush=True)
+  print(f'args: {args.namespace}')
```
Comment (mbobrovskyi) [inline src/xpk/commands/job.py:36]: Is it for debug? Can we remove it?

File: src/xpk/commands/job.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+def run_slurm_job_list_command(args) -> None:
+  cmd = (
+      f'kubectl-kjob list slurm  --profile {APP_PROFILE_TEMPLATE_DEFAULT_NAME}'
+  )
+
+  return_code = run_command_with_updates(cmd, 'list slurm jobs', args)
+  if return_code != 0:
+    xpk_print(f'Listing slurm jobs returned ERROR {return_code}')
+    xpk_exit(return_code)
```
Comment (mbobrovskyi) [inline src/xpk/commands/job.py:50]:
```suggestion
  xpk_exit(return_code)
```

File: src/xpk/commands/job.py
Comment (mbobrovskyi) [inline src/xpk/commands/job.py:1]: Do we need to add some info about this command on README?

Comment (IrvingMg) [reply src/xpk/commands/job.py:1]: README updated

File: README.md [hunk 6]
```diff
@@ -600,6 +600,24 @@ when creating the workload otherwise the workload will always finish with `Compl
     `125`: Workload finished but did not complete successfully.
     `1`: Other failure.
 
+## Job List
+
+*   Job List (see Slurm jobs):
```
Comment (mbobrovskyi) [inline README.md:605]: Where I can see? 🤔

Comment (IrvingMg) [reply README.md:605]: Maybe I misunderstood but the text in () is just a very short description of what Job List does.

Comment (IrvingMg) [reply README.md:605]: Sure. Should I remove it as well from the command description?

Comment (mbobrovskyi) [reply README.md:605]: I think so.

Comment (IrvingMg) [reply README.md]: Done.

Comment (mbobrovskyi) [reply README.md]: Thanks!

File: .github/workflows/build_tests.yaml [hunk 7]
```diff
@@ -88,6 +88,10 @@ jobs:
       run: python3 xpk.py workload delete --workload $WORKLOAD_NAME --cluster $TPU_CLUSTER_NAME --zone=us-central2-b
     - name: Delete the Pathways workload on the cluster
       run: python3 xpk.py workload delete --workload $PATHWAYS_WORKLOAD_NAME --cluster $TPU_CLUSTER_NAME --zone=us-central2-b
+    - name: Run a batch job on the cluster
+      run: python3 xpk.py batch --cluster $TPU_CLUSTER_NAME --zone=us-central2-b test.sh
+    - name: List out the jobs on the cluster
+      run: python3 xpk.py job ls | grep 'xpk-def-app-profile-slurm-'
```
Comment (mbobrovskyi) [inline .github/workflows/build_tests.yaml:94]: I think we should fail test if something not correct.

Comment (IrvingMg) [reply .github/workflows/build_tests.yaml:94]: As I understand, grep will exit with code 1 if the pattern `xpk-def-app-profile-slurm-'` is not found which will make the test failing

Comment (mbobrovskyi) [reply .github/workflows/build_tests.yaml:94]: Ah I see. Thanks!

**Review by mbobrovskyi — DISMISSED** [review]
LGTM. Thanks!

File: src/xpk/commands/job.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  """Function around job list.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    None
+  """
+  add_zone_and_project(args)
+  xpk_print(f'For project {args.project} and zone {args.zone}:', flush=True)
```
Comment (pawloch00) [inline src/xpk/commands/job.py:35]: Listing jobs for project ...

**Review by mbobrovskyi — APPROVED** [review]

**Review by pawloch00 — APPROVED** [review]

**Comment by IrvingMg** [comment]
Cluster deletion is failing in the pipeline. I think it might be because I need to remove the job created before deleting the cluster. The job deletion is done in #248. 

Maybe I should skip the test here and add both tests - ls and cancel - in #248? @pawloch00 WDYT?

**Comment by pawloch00** [comment]
> Cluster deletion is failing in the pipeline. I think it might be because I need to remove the job created before deleting the cluster. The job deletion is done in #248.
> 
> Maybe I should skip the test here and add both tests - ls and cancel - in #248? @pawloch00 WDYT?

I think cluster is upgrading and it has nothing to do with your PR. Please wait and try to run check, I will try to delete cluster manually.

**Comment by IrvingMg** [comment]
> > Cluster deletion is failing in the pipeline. I think it might be because I need to remove the job created before deleting the cluster. The job deletion is done in #248.
> > Maybe I should skip the test here and add both tests - ls and cancel - in #248? @pawloch00 WDYT?
> 
> I think cluster is upgrading and it has nothing to do with your PR. Please wait and try to run check, I will try to delete cluster manually.

Got it. Thanks!

**Pull request merged**